unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs

```diff
@@ -209,6 +210,15 @@ public void LoadUserProfile(UserProfile userProfile, bool forceLoading)
         if (userProfile.avatar == null || string.IsNullOrEmpty(userProfile.avatar.bodyShape))
             return;
 
+        /*TODO: this has to be refactored, currently there is no other way of understanding if the user is a regular or a guest
+        *       due to the execution order of things. The init cannot be done below because that would mean to do it when the
+        *       menu is firstly opened
+        */
+        if (!view.arePanelsInitialized)
+        {
+            view.InitializeNavigationEvents(string.IsNullOrEmpty(userProfile.userName));
+        }
```

Comment:
DataStore.i.common.isSignUpFlow.Get() To me returns false both as guest and normal user, I think it is because the initialization is done before the DataStore is correctly populated

Reply:
Ok! A last thing: We can remove this conditional as it is already being evaluated inside `InitializeNavigationEvents()` function, right?

```suggestion
        view.InitializeNavigationEvents(string.IsNullOrEmpty(userProfile.userName));
```